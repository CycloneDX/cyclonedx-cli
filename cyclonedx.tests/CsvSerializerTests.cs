using System;
using System.IO;
using Xunit;
using Snapshooter;
using Snapshooter.Xunit;
using CycloneDX.Xml;

namespace CycloneDX.CLI.Tests
{
    public class CsvSerializerTests
    {
        [Theory]
        [InlineData("bom")]
        [InlineData("valid-component-hashes")]
        [InlineData("valid-component-swid")]
        [InlineData("valid-component-swid-full")]
        [InlineData("valid-component-types")]
        [InlineData("valid-license-expression")]
        [InlineData("valid-license-id")]
        [InlineData("valid-license-name")]
        public void SerializationTests(string filename)
        {
            using (var tempDirectory = new TempDirectory())
            {
                var resourceFilename = Path.Join("Resources", filename + "-1.2.xml");
                var inputBomString = File.ReadAllText(resourceFilename);
                var bom = Xml.Deserializer.Deserialize(inputBomString);

                var bomCsv = CsvSerializer.Serialize(bom);

                Snapshot.Match(bomCsv, SnapshotNameExtension.Create(filename));
            }
        }

        [Theory]
        [InlineData("bom")]
        [InlineData("bom-minimum-viable")]
        [InlineData("bom-lowercase-field-names")]
        [InlineData("valid-component-hashes")]
        [InlineData("valid-component-swid")]
        [InlineData("valid-component-swid-full")]
        [InlineData("valid-component-types")]
        [InlineData("valid-license-expression")]
        [InlineData("valid-license-id")]
        [InlineData("valid-license-name")]
        public void DeserializationTests(string filename)
        {
            using (var tempDirectory = new TempDirectory())
            {
                var resourceFilename = Path.Join("Resources", filename + ".csv");
                var inputBomString = File.ReadAllText(resourceFilename);

                var bom = CsvSerializer.Deserialize(inputBomString);

                var bomXml = Xml.Serializer.Serialize(bom);

                Snapshot.Match(bomXml, SnapshotNameExtension.Create(filename));
            }
        }
    }
}
