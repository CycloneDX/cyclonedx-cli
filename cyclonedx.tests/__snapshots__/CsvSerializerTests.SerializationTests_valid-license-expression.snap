Type,MimeType,Supplier,Author,Publisher,Group,Name,Version,Scope,LicenseExpressions,LicenseNames,Copyright,Cpe,Purl,Modified,SwidTagId,SwidName,SwidVersion,SwidTagVersion,SwidPatch,SwidTextContentType,SwidTextEncoding,SwidTextContent,SwidUrl,MD5,SHA-1,SHA-256,SHA-512,BLAKE2b-256,BLAKE2b-384,BLAKE2b-512,SHA-384,SHA3-256,SHA3-384,SHA3-512,BLAKE3,Description
Application,,,,Acme Inc,com.acme,tomcat-catalina,9.0.14,required,EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0,,,,pkg:maven/com.acme/tomcat-catalina@9.0.14?packaging=jar,,,,,,,,,,,3942447fac867ae5cdb3229b658f4d48,e6b1000b94e835ffd37f4c6dcbdad43f4b48a02a,f498a8ff2dd007e29c2074f5e4b01a9a01775c3ff3aeaf6906ea503bc5791b7b,e8f33e424f3f4ed6db76a482fde1a5298970e442c531729119e37991884bdffab4f9426b7ee11fccd074eeda0634d71697d6f88a460dce0ac8d627a29f7d1282,,,,,,,,,Modified version of Apache Catalina
